core/src/Kokkos_Serial.hpp

```diff
@@ -634,6 +634,90 @@ class ParallelScan< FunctorType
     {}
 };
 
+/*--------------------------------------------------------------------------*/
+//VINH DANG -- Adding the following for returning final scan result
+template< class FunctorType , class ReturnType, class ... Traits >
+class ParallelScan_< FunctorType
```

Comment:
The underscore is strange and not meaningful. Could name this `ParallelScanWithTotal` (or something else meaningful) instead of `ParallelScan_`.